main.cpp

```diff
@@ -291,27 +350,75 @@ struct PaintballGun
     int chamber = 0;
     float oil = 1.0f;
     float valve = 0.5f;
+    bool triggerLock = false;
+
+    PaintballGun();
 
     bool shoot();
     bool load(); 
     void lock();
 };
 
+PaintballGun::PaintballGun()
+{
+    std::cout << "PaintballGun UDT being constructed!" << std::endl; //1)
+}
+
 bool PaintballGun::shoot()
 {
-    if( chamber == 1)
+    if (triggerLock)
+        {
+            std::cout << "Can't fire! Locked!" << std::endl;
+
+            return false;
+        }
+
+    else
     {
-        // shoot it
-        chamber = 0;
-        return true;
+        if( chamber == 1)
+        {
+            // shoot it
+            chamber = 0;
+            std::cout << "Ball shot! ------------->" << std::endl;
+
+            return true;
+        }
+
     }
```

Comment:
1) Fix this indentation
2) what do you return if `triggerLock` is `false`, and `chamber` is not `1` ?
3) If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise

Reply:
Cleaned this up a lot.